librosa/sequence.py

```diff
@@ -216,6 +216,11 @@ def dtw(X=None, Y=None, C=None, metric='euclidean', step_sizes_sigma=None,
                                       weights_mul, weights_add,
                                       max_0, max_1)
 
+    if subseq is False and np.isinf(D[-1, -1]):
```

Comment:
You can make this a one-liner with some Boolean fiddling:

`if np.isinf(D[-1, -1]) and (not subseq or np.all(np.isinf(D[-1, :])))`

Reply:
I found the two if-clauses more easy to understand. Have not a strong opinion in the case, though... Are there any librosa-guidelines for short code vs easy to understand code?

Reply:
I'm not a fan of strict guidelines here.  In this specific case, the if conditions are more readable as @fzalkow originally wrote them.  However, both conditions lead to the same (duplicated) code block (the runtime error), which isn't great style.  Because of that, I'd prefer @lostanlen's suggestion since it reduces branching paths and redundant code.

Reply:
tagging @bmcfee

Reply:
Ok, thanks for your input. Merged the lines and added a comment for understandability.